_drafts/gdbundle.md

```diff
@@ -0,0 +1,526 @@
+slowly grew it into a collection of scripts and improved our workflows. A few
+notable improvements were:
+
+- Add more scripts for each of our critical modules in the firmware, such as
+  state machines, Bluetooth state, etc.
+- Instead of just printing the heap allocations, expose the heap through a
+  Python Class. e.g.
+  `heap = Heap(g_heap); total_size = sum([b.size] for b in heap.iter_blocks()])`.
+- [Wrapping the GDB
+  invocation]({% post_url 2019-08-27-building-a-cli-for-firmware-projects %})
+  and loading GDB Python scripts automatically so every developer on the team
+  had them loaded and ready to use at all times.
+
+This was all back in 2015 or so. As I've moved around to other companies and
+talked to many other embedded software developers, I've discovered that many of
+them have never thought to extend or automate GDB using scripts, and even more
+had not even heard of GDB's Python API. I'm saddened by this.
+
+I have a personal goal (and it's one of Memfault's missions) to empower embedded
+software developers with the tools they need to get their job done, and I'm
+excited to share with you something I've been thinking about and building
+towards for over 3 years.
+
+<!-- excerpt start -->
+
+GDB desperately needs a better way for developers to share scripts,
+user-interface improvements, and utilities with the broader public. This would
+enable building upon each other's work and debugging techniques, progressing the
+entire community and preventing developers from reinventing the wheel time and
+time again. GDB (and LLDB) needs a plugin manager, and I'd like to introduce to
+you [gdbundle](https://github.com/memfault/gdbundle).
+
+<!-- excerpt end -->
+
+> This article speaks primarily to GDB, but gdbundle works perfectly
+> with LLDB as well.
+
+_Like Interrupt? [Subscribe](http://eepurl.com/gpRedv) to get our latest posts
+straight to your mailbox_
+
+{:.no_toc}
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
+
+## Why Does GDB Need a Plugin Manager
+
+What has happened in the GDB ecosystem due to not having a plugin manager?
+
+### No Out-Of-Box Debugging of Popular Software
+
+Within this past year, I've worked with four different microcontroller stacks, 7
+different Real-Time Operating Systems (RTOS's), and a handful of common
+low-level software libraries including Mbed TLS, the WICED Bluetooth Stack, and
+many mediocre vendor SDK's. Tens of thousands of developers use each one of
+these stacks/libraries, and every single one of them has to manually debug each
+module by hand by using GDB's print functionality or write their own scripts.
+
+This is why embedded developers often choose to use thousand dollar or mediocre
+IDE debuggers over using GDB. They have these debugging utilities built in or
+allow extensions to be integrated and sold[^code_confidence], even though the
+software backing them isn't all that complex.
+
+### Poor Installation and Non-Discoverable of Extensions
```

Comment:
I would significantly rework this sub point so it reads more like "something we're missing out on" rather than "other people do it so it must be good".

I'd go with this:
```
### Package Managers Build Community
```
and talk about how VSCode, and even PlatformIO have built a big community of people around the exchange & use of plugins.